binding/python/test_rhino.py

```diff
@@ -39,77 +46,133 @@ def _context_path():
-                                        '../../resources/contexts/raspberry-pi/coffee_maker_raspberry-pi.rhn')
+                                        contexts_root, 'raspberry-pi', f'{context}_raspberry-pi.rhn')
                 elif '0xd07' == cpu_part:
                     return os.path.join(os.path.dirname(__file__),
-                                        '../../resources/contexts/jetson/coffee_maker_jetson.rhn')
+                                        contexts_root, 'jetson', f'{context}_jetson.rhn')
                 elif '0xc08' == cpu_part:
                     return os.path.join(os.path.dirname(__file__),
-                                        '../../resources/contexts/beaglebone/coffee_maker_beaglebone.rhn')
+                                        contexts_root, 'beaglebone', f'{context}_beaglebone.rhn')
                 else:
                     raise NotImplementedError("Unsupported CPU: '%s'." % cpu_part)
         elif system == 'Windows':
-            return os.path.join(os.path.dirname(__file__), '../../resources/contexts/windows/coffee_maker_windows.rhn')
+            return os.path.join(os.path.dirname(__file__), contexts_root, 'windows', f'{context}_windows.rhn')
         else:
             raise ValueError("Unsupported system '%s'." % system)
 
-    rhino = None
+    @classmethod
+    def __pv_model_path_by_language(cls, relative, language):
+        model_path_subdir = cls.__append_language('lib/common/rhino_params', language)
+        model_path_subdir = f'{model_path_subdir}.pv'
+        return os.path.join(os.path.dirname(__file__), relative, model_path_subdir)
+
+    rhinos = None
 
     @classmethod
     def setUpClass(cls):
-        cls.rhino = Rhino(
-            access_key=sys.argv[1],
-            library_path=pv_library_path('../..'),
-            model_path=pv_model_path('../..'),
-            context_path=cls._context_path())
+        _language_to_contexts = {
+            'en': ['coffee_maker'],
+            'es': ['luz'],
+            'de': ['beleuchtung']
+        }
+
+        cls.rhinos = dict()
+        for language in _language_to_contexts:
+            cls.rhinos[language] = dict()
+            for context in _language_to_contexts[language]:
+                cls.rhinos[language][context] = Rhino(
+                    access_key=sys.argv[1],
+                    library_path=pv_library_path('../..'),
+                    model_path=cls.__pv_model_path_by_language('../..', language),
+                    context_path=cls.__context_path(context, language)
+                )
 
     @classmethod
     def tearDownClass(cls):
-        if cls.rhino is not None:
-            cls.rhino.delete()
+        if cls.rhinos is not None:
+            for language in cls.rhinos:
+                for context in cls.rhinos[language]:
+                    cls.rhinos[language][context].delete()
+
+    def run_rhino(self, language, context, is_whithin_context, intent=None, slots=None,
```

Comment:
it looks like `is_within_context` decides which audio file to use? I think we should remove that argument and just explicitly pass in whatever audio file name we're looking to use.

Reply:
actually `is_within_context` it is used to determine whether the input audio file is within the given context or not.
Ali said to automatically generate the audio name if possible, so I also used it there. I'll revert and pass the audio files manually again.

Reply:
oh okay, my bad then. Either is fine then.